### PR TITLE
Documentation: MPI_Info_set string arguments are const

### PR DESCRIPTION
**Pull request by devreal**
Signed-off-by: Joseph Schuchart <schuchart@icl.utk.edu>

### CODE REVIEW

**Review by jsquyres — APPROVED** [review]
Awesome; thanks.  Are the other Info man pages correct?

**Comment by devreal** [comment]
Indeed, the `datarep` argument to `MPI_Pack_external_size` was missing `const` too. I didn't find any other instances.

File: docs/man-openmpi/man3/MPI_Info_set.3.rst [hunk 1]
```diff
@@ -20,7 +20,7 @@ C Syntax
 
    #include <mpi.h>
 
-   int MPI_Info_set(MPI_Info info, char *key, char *value)
+   int MPI_Info_set(MPI_Info info, const char *key, const char *value)
```
Comment (awlauria) [inline docs/man-openmpi/man3/MPI_Info_set.3.rst:23]: Out side the scope of this pr - why are some of these in italics and some of them are not? Seems random.

**Comment by devreal** [comment]
Thanks! Backport to v5.0.x in https://github.com/open-mpi/ompi/pull/10275 (thanks @awlauria). Should the same go into v4.1.x and v4.0.x (on the old man files)?

**Pull request merged**